src/core/reference/src/utils/phillox_converter.cpp

```diff
@@ -0,0 +1,513 @@
+        convert_to_output_type<int>(result,
+                                    m_uints_generated_per_exec,
+                                    get_converted_elements_count(),
+                                    m_elem_type,
+                                    m_min_val,
+                                    m_max_val,
+                                    m_out,
+                                    idx,
+                                    m_elem_count,
+                                    nullptr,
+                                    nullptr,
+                                    [](uint32_t x, int mn, int mx) {
+                                        return static_cast<int>(x % (mx - mn) + mn);
+                                    });
+        break;
+    }
+    case element::Type_t::i64: {
+        convert_to_output_type<int64_t>(result,
+                                        m_uints_generated_per_exec,
+                                        get_converted_elements_count(),
+                                        m_elem_type,
+                                        m_min_val,
+                                        m_max_val,
+                                        m_out,
+                                        idx,
+                                        m_elem_count,
+                                        nullptr,
+                                        [](uint32_t a, uint32_t b, int64_t mn, int64_t mx) {
+                                            return static_cast<int64_t>(unite_high_low(b, a) % (mx - mn) + mn);
+                                        });
+        break;
+    }
+    default:
+        OPENVINO_THROW("Unsupported type of RandomUniform: ", m_elem_type.to_string());
+    }
+}
+
+// ======= PyTorchPhilloxConverter functions =======
+
+PyTorchPhilloxConverter::PyTorchPhilloxConverter(char* out,
+                                                 const element::Type& elem_type,
+                                                 const size_t elem_count,
+                                                 const char* min_val,
+                                                 const char* max_val,
+                                                 const size_t uints_generated_per_exec)
+    : PhilloxConverter(out, elem_type, elem_count, min_val, max_val, uints_generated_per_exec) {}
+
+size_t PyTorchPhilloxConverter::get_converted_elements_count() const {
+    // PyTorch uses one uint32_t value per generated output
+    // except for int64 only
+    if (m_elem_type.size() > 4) {
+        int64_t mn[1];
+        int64_t mx[1];
+        memcpy(mn, m_min_val, m_elem_type.size());
+        memcpy(mx, m_max_val, m_elem_type.size());
+
+        if (mn[0] <= std::numeric_limits<uint32_t>::max() && mx[0] <= std::numeric_limits<uint32_t>::max()) {
+            return 4;
+        }
+        return 2;
+    }
+    return 4;
+    // return m_elem_type.size() > 4 ? 2 : 4;
+    // return m_elem_type == element::i64 ? 2 : 4;
+}
```

Comment:
Commented out code should be removed:
```suggestion
    return 4;
}
```

Reply:
Removed!